flink-table/flink-table-planner-blink/src/test/resources/org/apache/flink/table/plan/batch/sql/join/LookupJoinTest.xml

```diff
@@ -233,6 +233,34 @@ FlinkLogicalAggregate(group=[{0}], EXPR$1=[COUNT($1)], EXPR$2=[SUM($2)], EXPR$3=
       +- FlinkLogicalSnapshot(period=[$cor0.proctime])
          +- FlinkLogicalCalc(select=[id], where=[>(age, 10)])
             +- FlinkLogicalTableSourceScan(table=[[default_catalog, default_database, temporalTest, source: [TestTemporalTable(id, name, age)]]], fields=[id, name, age])
+]]>
+    </Resource>
+  </TestCase>
+  <TestCase name="testJoinTemporalTableWithTrueCondition">
+    <Resource name="sql">
+      <![CDATA[
+SELECT * FROM MyTable AS T
+JOIN temporalTest FOR SYSTEM_TIME AS OF T.proctime AS D
+ON true
```

Comment:
This is a feature you just added or it is caused by upgrading calcite to 1.20?

Reply:
before 1.20, the filter will not be removed even the join condition is `true`. while after [CALCITE-2004](https://issues.apache.org/jira/browse/CALCITE-2004) is fixed in 1.20, the filter will be removed when join condition is `true`

Reply:
add some comments to explain?

Reply:
BTW, could you add some cases for CALCITE-2004, it seems as a serious bug and it also reflected we don't have such test case

Reply:
currently, both blink planner and flink planner choose a temporary solution: only allow true condition, otherwise throw exception (see: [FLINK-7865](https://issues.apache.org/jira/browse/FLINK-7865)). should we remove the join validation to support CALCITE-2004 in 1.9 ?